lib/yunohost/tools.py

```diff
@@ -469,3 +474,44 @@ def tools_upgrade(auth, ignore_apps=False, ignore_packages=False):
     if is_api:
         from yunohost.service import service_log
         return { "log": service_log('yunohost-api', number="100").values()[0] }
+
+
+def tools_upgrade_v24(auth):
+    """
+    YunoHost upgrade to new Yunohost version (on jessie)
+    """
+    import platform
```

Comment:
It's better yes. The cases where it's kinda ok to put import somewhere else is when you do that in a "except" block where the import is guaranteed not to fail.